### PR TITLE
State crawler jobs from cli

### PR DESCRIPTION
**Pull request by Andrei-Dolgolev**
## Changes

Add reading jobs from file.
Add Thread with timeout for pass out node hung connection.
Add batch size auto-scale up to 5000.
Fix tree parsing.
Add more logging.
Add ability set custom web3 http provider for be able use different providers.
Add INFURA_PROJECT_ID as env variable.

### Custom Web3 provider command:
```
state-crawler --access-id $NB_CONTROLLER_ACCESS_ID crawl-jobs --blockchain polygon --custom-web3-provider $WEB3_POLYGON --jobs-file ./mooncrawl/state_crawler/tasks.json
```

### **Infura** project command:
**polygon**
```
state-crawler --access-id $NB_CONTROLLER_ACCESS_ID crawl-jobs --blockchain polygon --infura --jobs-file ./mooncrawl/state_crawler/jobs/polygon-jobs.json
```

**mumbai**
```
state-crawler --access-id $NB_CONTROLLER_ACCESS_ID crawl-jobs --blockchain mumbai --infura --jobs-file ./mooncrawl/state_crawler/jobs/mumbai-jobs.json
```

Currently user supose track correct contract and view methods with correct blockchain.

### CODE REVIEW

**Review by zomglings — APPROVED** [review]
lgtm

File: crawlers/deploy/polygon-state.service [hunk 1]
```diff
@@ -8,6 +8,6 @@ User=ubuntu
 Group=www-data
 WorkingDirectory=/home/ubuntu/moonstream/crawlers/mooncrawl
 EnvironmentFile=/home/ubuntu/moonstream-secrets/app.env
-ExecStart=/home/ubuntu/moonstream-env/bin/python -m mooncrawl.state_crawler.cli --access-id "${NB_CONTROLLER_ACCESS_ID}" crawl-jobs --blockchain polygon
+ExecStart=/home/ubuntu/moonstream-env/bin/python -m mooncrawl.state_crawler.cli --access-id "${NB_CONTROLLER_ACCESS_ID}" crawl-jobs --blockchain polygon --infura --jobs-file ./mooncrawl/state_crawler/jobs/polygon-jobs.json
```
Comment (kompotkot) [inline crawlers/deploy/polygon-state.service:11]: `--jobs-file ./mooncrawl/state_crawler/jobs/polygon-jobs.json`
to
`--jobs-file /home/ubuntu/moonstream/crawlers/mooncrawl/state_crawler/jobs/polygon-jobs.json`

Comment (kompotkot) [reply crawlers/deploy/polygon-state.service:11]: better to use full path in systemd files

Comment (kompotkot) [reply crawlers/deploy/polygon-state.service:11]: or env variable

File: crawlers/mooncrawl/mooncrawl/state_crawler/cli.py [hunk 2]
```diff
@@ -13,38 +17,51 @@
...
-    multicall_calls = [
-        (
-            call["address"],
-            call["method"].encode_data(call["inputs"]).hex(),
-        )
-        for call in calls
-    ]
+    multicall_calls = []
+
+    # Remove!
```
Comment (kompotkot) [inline crawlers/mooncrawl/mooncrawl/state_crawler/cli.py:50]: remove?

File: crawlers/mooncrawl/mooncrawl/state_crawler/cli.py [hunk 3]
```diff
@@ -183,9 +274,26 @@ def parse_jobs(
...
+            logger.info(
+                f"Connecting to blockchain: {blockchain_type} with custom provider!"
+            )
+
+            web3_client = connect(web3_provider_uri)
+
+            if blockchain_type != AvailableBlockchainType.ETHEREUM:
+                web3_client.middleware_onion.inject(geth_poa_middleware, layer=0)
+        except Exception as e:
+            logger.info(
```
Comment (kompotkot) [inline crawlers/mooncrawl/mooncrawl/state_crawler/cli.py:288]: logger.error

**Review by kompotkot — APPROVED** [review]

**Pull request merged**